docs/rest_api_guide/02_RESTauthentication.rst

```diff
@@ -0,0 +1,365 @@
+
+``PUT /fledge/{user_id}/logout`` - Terminate the login session for another user.
+
+The administrator may terminate the login session of another user.
+
+.. code-block:: console
+
+   curl -H "authorization: <token>" -X PUT http://localhost:8081/fledge/{user_id}/logout
+
+Users
+=====
+
+Fledge supports two levels of user, administration users and normal users. A set of API calls exists to allow users to be created, queried, modified and destroyed. 
+
+Add User
+--------
+
+``POST /fledge/user`` - add a new user to Fledge’s user database
+
+**Request Payload**
+
+A JSON document which describes the user to add.
+
+.. list-table::
+    :widths: 20 20 50 30
+    :header-rows: 1
+
+    * - Name
+      - Type
+      - Description
+      - Example
+    * - username
+      - string
+      - The username of the new user to add
+      - david
+    * - password
+      - string
+      - The password to assign to the new user. If not given then a certificate must be included in the payload.
+      - 1nv1nc1ble
+    * - certificate
+      - string
+      - The name of a certificate in the certificate store. May only be used when a password is not given.
+      -
+    * - permissions
+      - string
+      - The permissions that new user should be given
+      - admin
+    * - realname
+      - string
+      - The real name of the user. This is used for display purposes only.
+      - David Brent
+
+**Response Payload**
+
+The response payload is a JSON document containing the username of the newly created user.
+
+**Errors**
+
+The following error responses may be returned
+
+.. list-table::
+    :widths: 20 50
+    :header-rows: 1
+
+    * - HTTP Code
+      - Reason
+    * - 400
+      - Incomplete or badly formed request payload
+    * - 403
+      - A user without admin permissions tried to add a new user
+    * - 409
+      - The username is already in use
+
+
+**Example**
+
+.. code-block:: console
+
+   curl -X POST /fledge/user -d'
+   {
+    "username"    : "david",
+    "password"    : "1nv1nc1ble",
+    "permissions" : "admin",
+    "realname"    : "David Brent"
+   }
+
+Get All Users
+-------------
+
+``GET /fledge/user`` - Retrieve data on all users
+
+**Response Payload**
+
+A JSON document which all users in a JSON array.
+
+.. list-table::
+    :widths: 20 20 50 30
+    :header-rows: 1
+
+    * - Name
+      - Type
+      - Description
+      - Example
+    * - [].username
+      - string
+      - The username of the new user to add
+      - david
+    * - [].permissions
+      - string
+      - The permissions that new user should be given
+      - admin
+    * - [].realname
+      - string
+      - The real name of the user. This is used for display purposes only.
+      - David Brent
+
+.. note::
+
+   This payload does not include the password of the user.
+
+**Example**
+
+.. code-block:: console
+
+   curl -X GET /fledge/user
+
+
+Returns the response payload
+
+.. code-block:: console
```

Comment:
`.. code-block:: json`
Also add blank line so that no errors appear on code-block directive.